includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -922,6 +923,15 @@ private function validate_cdata_for_node( DOMElement $element, $cdata_spec ) {
 				return [ 'code' => self::MANDATORY_CDATA_MISSING_OR_INCORRECT ];
 			}
 		}
+
+		// When the CDATA is expected to be JSON, ensure it's valid JSON.
+		if ( 'script' === $element->tagName &&
+			in_array( $element->getAttribute( 'type' ), [ 'application/json', 'application/ld+json' ], true ) &&
+			null === json_decode( $element->textContent )
```

Comment:
This isn't quite right. What if the JSON being decoded is itself `null`? This needs to make use of `json_last_error()` to detect the error scenario.

Reply:
OK, sure

Reply:
https://github.com/ampproject/amp-wp/pull/4340/commits/ea26570a49ca6f3cbfc8a32c7e64aab3a4a91803 uses `json_last_error()` instead.